radix-engine-profiling/src/rocks_db_metrics/tests/delete.rs

```diff
@@ -0,0 +1,463 @@
+    export_graph_and_print_summary(
+        &format!(
+            "RocksDB per partition deletion with JMT (N=1..{}) rounds: {}",
+            N, ROUNDS_COUNT
+        ),
+        &jmt_rocksdb_data,
+        &jmt_rocksdb_data_output,
+        "/tmp/scrypto_delete_per_partition_rocksdb_JMT.png",
+        "95th percentile of deletion",
+        &jmt_rocksdb_data_original,
+        axis_ranges,
+        Some("N"),
+    )
+    .unwrap();
+
+    export_graph_two_series(
+        &format!(
+            "95th percentile of deletion per partition, rounds: {}",
+            ROUNDS_COUNT
+        ),
+        &rocksdb_data_output,
+        &jmt_rocksdb_data_output,
+        "/tmp/scrypto_delete_per_partition_rocksdb_diff.png",
+        "N",
+        "Duration [µs]",
+        "Series 1: deletion",
+        "Series 2: deletion with JMT",
+        (lin_slope, lin_intercept),
+        (jmt_lin_slope, jmt_lin_intercept),
+    )
+    .unwrap();
+}
+
+fn test_delete_per_size_internal<F, S>(
+    rounds_count: usize,
+    min_size: usize,
+    max_size: usize,
+    size_step: usize,
+    prepare_db_write_repeats: usize,
+    create_store: F,
+) -> (
+    Vec<(f32, f32)>,
+    Vec<(f32, f32)>,
+    BTreeMap<usize, Vec<Duration>>,
+)
+where
+    F: Fn(PathBuf) -> SubstateStoreWithMetrics<S>,
+    S: SubstateDatabase + CommittableSubstateDatabase,
+{
+    // RocksDB part
+    let path = PathBuf::from(r"/tmp/radix-scrypto-db");
+    // clean database
+    std::fs::remove_dir_all(path.clone()).ok();
+
+    // prepare database
+
+    // Stage 1: fill db with substates to prepare base db size
+    // this will prevent from decreasing db size to 0 when deleting substates
+    {
+        let mut substate_db = create_store(path.clone());
+
+        prepare_db(
+            &mut substate_db,
+            min_size,
+            max_size,
+            size_step,
+            prepare_db_write_repeats,
+        );
+    }
+
+    // Stage 2: reopen database and fill db with additional substates which will be deleted in next step
+    let data: Vec<(DbPartitionKey, DbSortKey, usize)> = {
+        let mut substate_db = create_store(path.clone());
+
+        prepare_db(
+            &mut substate_db,
+            min_size,
+            max_size,
+            size_step,
+            prepare_db_write_repeats,
+        )
+    };
+
+    // reopen database and measure deletion times
+    let mut substate_db = create_store(path.clone());
+
+    println!("Delete test execution");
+    let mut rng = rand::thread_rng();
+
+    // repeat 1 substate commit n-times
+    for i in 0..rounds_count {
+        print!("Round {}/{}\r", i + 1, rounds_count);
+        std::io::stdout().flush().ok();
+
+        // prepare vector with indices of data to draw from
+        let mut size_vector: Vec<usize> = Vec::new();
+        for j in (i..data.len()).step_by(prepare_db_write_repeats) {
+            size_vector.push(j);
+        }
```

Comment:
This is very convoluted. Possible to have `data.shuffle()` and delete all the entries?

Reply:
Refactored all vectors randomizations to use `shuffle()` function.